posit-dev/positron
# [python] install Python into Conda environments on discovery

### Intent

> Describe briefly what problem this pull request resolves, or what new feature it introduces. Include screenshots of any new or altered UI. Link to any GitHub issues that are related.

addresses #2504 

### Approach

> Describe the approach taken and the tradeoffs involved if non-obvious; add an overview of the solution if it's complicated.

When a conda environment is added to Positron runtimes, if Python is NOT installed, install it. This follows VSCode's standards of installing Python into conda runtimes. 

I originally had the install on activation, but it doesn't feel as seamless since the problematic environment shows as `Python 0.0.1 (Conda: python)` in the Positron interpreter selector. If this feels too slow/is slowing down interpreter discovery, I'd be interested in discussing when this should occur.

I also added feedback if you enter an interpreter path that does not exist in the Command Palette, since currently the failure is silent.

Still TODO is fix failing tests and add tests.

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues.

In Terminal:

```
conda create -n testconda1234
```

(at `Proceed ([y]/n)? ` -> `y`)

Then you should see the environment in the Positron Interpreter Selector (last resort is doing `cmd+shift+p` -> `Select Interpreter` -> find `testconda1234` but I pretty reliably can find it in the Positron dropdown; both paths should have same behavior).

<img width="449" alt="Screenshot 2024-05-14 at 6 31 38 PM" src="https://github.com/posit-dev/positron/assets/54685329/c2780a76-9498-41ac-9cbe-c90acb1e729e">

You should be able to select this interpreter and it prompts you for IPykernel, which should succeed.
